lib/krane/cluster_resource_discovery.rb

```diff
@@ -23,9 +23,19 @@ def prunable_resources(namespaced:)
       fetch_resources(namespaced: namespaced).uniq { |r| r['kind'] }.map do |resource|
         next unless resource['verbs'].one? { |v| v == "delete" }
         next if black_list.include?(resource['kind'])
-        group_versions = api_versions[resource['apigroup'].to_s]
-        version = version_for_kind(group_versions, resource['kind'])
-        [resource['apigroup'], version, resource['kind']].compact.join("/")
+        group = resource['apigroup'].to_s
+        version = ''
+        if group.empty?
```

Comment:
I think this logic should be pushed into the version_for_kind function or at least not this function.